src/toniarts/openkeeper/tools/convert/map/KwdFile.java

```diff
@@ -142,48 +144,50 @@ public final class KwdFile {
 
     /**
      * Constructs a new KWD file reader<br>
-     * Reads the whole map and its catalogs (either standard ones or custom
-     * ones)
+     * Reads the whole map and its catalogs (either standard ones or custom ones)
      *
      * @param basePath path to DK II main path (or where ever is the "root")
-     * @param file the KWD file to read
+     * @param name the KWD file to read
+     * @throws java.io.IOException if file not exists
      */
-    public KwdFile(String basePath, Path file) {
-        this(basePath, file, true);
+    public KwdFile(String basePath, String name) throws IOException {
+        this(basePath, name, true);
     }
 
     /**
      * Constructs a new KWD file reader<br>
      *
      * @param basePath path to DK II main path (or where ever is the "root")
-     * @param file the KWD file to read
-     * @param load whether to actually load the map data, or just get the
-     * general info
+     * @param name the KWD file to read
+     * @param load whether to actually load the map data, or just get the general info
+     * @throws java.io.IOException if file not exists
      */
-    public KwdFile(String basePath, Path file, boolean load) {
+    public KwdFile(String basePath, String name, boolean load) throws IOException {

```

Comment:
I think this now goes backwards. KwdFile is just reader/parser for KWD files. It should not have any logic as to where the files are. I see that you might want to stop the repeating code and move the logic elsewhere, that is ok. But the logic is not here.

Either util method that does it or some factory that does it. But KwdFile should not know it, just a reader.